inlang/source-code/paraglide/paraglide-js/src/compilerV2/compileMessage.test.ts

```diff
@@ -0,0 +1,64 @@
+import { describe, it, expect } from "vitest"
+import { compileMessage } from "./compileMessage.js"
+import type { MessageNested } from "@inlang/sdk2"
+
+describe("compileMessage", () => {
+	it("compiles a message with variants", () => {
+		const msg: MessageNested = {
+			locale: "en",
+			id: "some_message",
+			bundleId: "some_bundle",
+			declarations: [
+				{
+					type: "input",
+					name: "fistInput",
+					value: { type: "expression", arg: { type: "variable", name: "fistInput" } },
+				},
+				{
+					type: "input",
+					name: "secondInput",
+					value: { type: "expression", arg: { type: "variable", name: "second Input" } },
+				},
+			],
+			selectors: [
+				{ type: "expression", arg: { type: "variable", name: "fistInput" } },
+				{
+					type: "expression",
+					arg: { type: "variable", name: "second Input" },
+					annotation: { type: "function", name: "plural", options: [] },
+				},
+			],
+			variants: [
+				{
+					id: "1",
+					messageId: "some_message",
+					match: ["1", "2"],
+					pattern: [{ type: "text", value: "One" }],
+				},
+				{
+					id: "2",
+					messageId: "some_message",
+					match: ["*", "*"],
+					pattern: [{ type: "text", value: "Many" }],
+				},
+			],
+		}
+
+		const compiled = compileMessage(msg)
+		expect(compiled.typeRestrictions).toEqual({
+			"second Input": "number",
+		})
+		expect(compiled.code).toMatchInlineSnapshot(`
+			"/**
+			 * @param {{ 'second Input': number, fistInput: NonNullable<unknown> }} inputs
+			 * @returns {string}
+			 */
+			/* @__NO_SIDE_EFFECTS__ */
+			const some_message = (inputs) => {
+				const selectors = [ inputs.fistInput, registry.plural(\\"en\\", inputs['second Input']) ]
```

Comment:
(no action needed)

beautiful comprehension. reading `registry.plural` immediately makes click "aha this function is coming from the registry".